src/main/java/com/teragrep/cfe_39/consumers/kafka/DatabaseOutput.java

```diff
@@ -0,0 +1,396 @@
+    private SyslogAvroWriter syslogAvroWriter;
+    private final long minimumFreeSpace;
+    private final long maximumFileSize;
+    private final WritableQueue writableQueue;
+    private final ByteBuffer sourceConcatenationBuffer;
+    private final SDVector teragrepStreamName;
+    private final SDVector teragrepDirectory;
+    private final SDVector eventNodeSourceSource;
+    private final SDVector eventNodeRelaySource;
+    private final SDVector eventNodeSourceSourceModule;
+    private final SDVector eventNodeRelaySourceModule;
+    private final SDVector eventNodeSourceHostname;
+    private final SDVector eventNodeRelayHostname;
+    private final SDVector originHostname;
+    private File syslogFile;
+    private final Config config;
+    private long approximatedSize; // FIXME: Not working properly when flush() is not used after append in the AVRO-file.
+    private long epochMicros_last;
+
+    DatabaseOutput(
+            Config config,
+            String table,
+            DurationStatistics durationStatistics,
+            TopicCounter topicCounter
+    ) {
+        this.config = config;
+        this.table = table;
+        this.durationStatistics = durationStatistics;
+        this.topicCounter = topicCounter;
+        this.minimumFreeSpace = 32000000; // TODO: CHECK RIGHT VALUE FOR minimumFreeSpace
+        this.maximumFileSize = config.getMaximumFileSize();; // Maximum file size should be 64M (64000000). 60800000 is 95% of 64M which should be a good approximation point.
+
+        // queueDirectory and queueNamePrefix are only used for temporarily storing the AVRO-serialized files before committing them to HDFS when the file size reaches the threshold (or all records are processed).
+        this.writableQueue = new WritableQueue(
+                config.getQueueDirectory()
+        );
+
+        this.sourceConcatenationBuffer = ByteBuffer.allocateDirect(256*1024);
+        teragrepStreamName = new SDVector("teragrep@48577","streamname");
+        teragrepDirectory = new SDVector("teragrep@48577","directory");
+        this.eventNodeSourceSource = new SDVector("event_node_source@48577","source");
+        this.eventNodeRelaySource = new SDVector("event_node_relay@48577","source");
+        this.eventNodeSourceSourceModule = new SDVector("event_node_source@48577","source_module");
+        this.eventNodeRelaySourceModule = new SDVector("event_node_relay@48577","source_module");
+        this.eventNodeSourceHostname = new SDVector("event_node_source@48577","hostname");
+        this.eventNodeRelayHostname = new SDVector("event_node_relay@48577","hostname");
+        this.originHostname = new SDVector("origin@48577","hostname");
+        this.approximatedSize = 0;
+        this.epochMicros_last = 0L;
+    }
+
+    // Checks that the filesize stays under the defined maximum file size.
+    boolean checkSizeTooLarge(long fileSize, RecordOffsetObject recordOffsetObject) {
```

Comment:
This method could be renamed to something else to improve readability, since it's also doing write operations in addition to doing the  fileSize check